src/experiments/index.js

```diff
@@ -142,42 +135,34 @@ export function experimentToggles(win) {
     }
   }
   // Read document level override from meta tag.
-  if (
-    win.AMP_CONFIG &&
-    Array.isArray(win.AMP_CONFIG['allow-doc-opt-in']) &&
-    win.AMP_CONFIG['allow-doc-opt-in'].length > 0
-  ) {
-    const allowed = win.AMP_CONFIG['allow-doc-opt-in'];
+  const allowedDocOptIn = win.AMP_CONFIG?.['allow-doc-opt-in'];
+  if (isArray(allowedDocOptIn) && allowedDocOptIn.length) {
     const meta = win.document.head.querySelector(
       'meta[name="amp-experiments-opt-in"]'
     );
     if (meta) {
       const optedInExperiments = meta.getAttribute('content').split(',');
-      for (let i = 0; i < optedInExperiments.length; i++) {
-        if (allowed.indexOf(optedInExperiments[i]) != -1) {
-          toggles[optedInExperiments[i]] = true;
+      for (const experiment of optedInExperiments) {
+        if (dev().assertArray(allowedDocOptIn).includes(experiment)) {
+          toggles[experiment] = true;
         }
       }
     }
   }
 
   Object.assign(toggles, getExperimentToggles(win));
 
-  if (
-    win.AMP_CONFIG &&
-    Array.isArray(win.AMP_CONFIG['allow-url-opt-in']) &&
-    win.AMP_CONFIG['allow-url-opt-in'].length > 0
-  ) {
-    const allowed = win.AMP_CONFIG['allow-url-opt-in'];
+  const allowedUrlOptIn = win.AMP_CONFIG?.['allow-url-opt-in'];
+  if (isArray(allowedUrlOptIn) && allowedUrlOptIn.length) {
```

Comment:
nit: are there things other than arrays that have length? Can it be condensed to: `win.AMP_CONFIG['allow-url-opt-in']?.length`

Reply:
If this property was assigned a string value, `length` could be truthy but for...of would have weird results